DOC: update docstring in anderson to reflect name of returned variable

TomDonoghue opened the pull request:
This is minor update on the documentation for the 'anderson' function docstring. 

Fixes:
- In the current docstring for the function, the note refers to the returned statistic as 'A2'. Though this is what the variable is called in the code, it is not mentioned that this stat (the first returned item) is what 'A2' refers to anywhere else in the docstring, and so looking only at the docs is a bit confusing, especially for the new user / one unfamiliar with the test. Here, simply replace the reference to 'A2' with 'statistic', as it is otherwise referred to in the docs, for consistency.

josef-pkt commented:
sounds almost fine to me.
Given that the description doesn't use `A2`, I agree that the rephrasing with `statistics` is better.
However, all the references refer to Stephens, and in many articles he has several statistics at the same time.
So, maybe adding some extra `A2 in Stephens` would be helpful for users that quickly want to check the references.

Earlier in the docstring, defining Anderson Darling as a modification of kstest doesn't sound quite right. They are similar, one uses max or sup, the other one a weighted  integral/average (squared).

TomDonoghue commented:
@josef-pkt 
I updated the docs to note that the statistics is referred to as 'A2' in the references, and removed the explicit comparison to Anderson as a version of kstest, and added it instead to the 'See Also' section. 

If a more nuanced note about how Anderson relates to kstest is warranted, I'm not sure I'm the best person to add that, but I'm happy to take any suggestions of how to frame that.
